### PR TITLE
fix: Default external-unstable ListenerClass back to IPs

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

Reasoning is on the code comments and changelog.
Relevant discussion: https://stackable-workspace.slack.com/archives/C0312UB3LLE/p1730312706917569?thread_ts=1730298381.528779&cid=C0312UB3LLE

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] CRD documentation for all fields, following the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs/style-guide).
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
- [ ] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] Code contains useful logging statements
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs/style-guide).
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
- [ ] [Roadmap](https://github.com/orgs/stackabletech/projects/25/views/1) has been updated
```

### CODE REVIEW

File: deploy/helm/listener-operator/templates/listener-classes.yaml [hunk 1]
```diff
@@ -51,6 +61,10 @@ spec:
   # or on-premise environments that don't support external LoadBalancer peering (such as Calico (https://docs.tigera.io/calico/latest/networking/configuring/advertise-service-ips)
   # or MetalLB (https://metallb.org/)).
   serviceType: NodePort
+  # As the nodes have stable addresses we expect them to have hostnames that are resolvable outside of Kubernetes.
+  # And yes, we know this causes problems when running on kind with the stable-nodes preset (default), as the kind node
+  # hostname is very likely not resolvable on the host machine.
+  preferredAddressType: Hostname
```
Comment (nightkr) [inline deploy/helm/listener-operator/templates/listener-classes.yaml:67]: This should poobably match external-unstable

File: deploy/helm/listener-operator/templates/listener-classes.yaml [hunk 2]
```diff
@@ -36,6 +41,11 @@ metadata:
   name: external-unstable
 spec:
   serviceType: NodePort
+  # Ideally we would prefer the Hostname here, but up to (including) SDP 24.7 we defaulted to the IP.
+  # We actually defaulted to Hostname for some weeks, but noticed this breaks external access to kind clusters, as the
+  # local machines can not resolve the hostnames of the nodes (such as kind-control-plane).
+  # So we stick to the old default and re-visit the default later - after SDP 24.11 is released.
+  preferredAddressType: IP
```
Comment (nightkr) [inline deploy/helm/listener-operator/templates/listener-classes.yaml:48]: This makes it sound like a temporary migration measure, but I suspect pretty heavily that we'll want to make this permanent.

**Review by nightkr — COMMENTED** [review]
Hm, another alternative would be to add some "auto" mode to preferredAddressType that reverts to the old behaviour, so we avoid the breakage entirely.

**Comment by sbernauer** [comment]
Superseded by https://github.com/stackabletech/listener-operator/pull/244

**Pull request closed without merging**